Source/WebKit/UIProcess/PageClient.h

```diff
@@ -195,7 +196,6 @@ class WebContextMenuProxy;
 class WebEditCommandProxy;
 class WebFrameProxy;
 class WebOpenPanelResultListenerProxy;
-class WebPageProxy;
```

Comment:
I had to do this because this `class WebPageProxy` forward declaration doesn't allow me to use `WebPageProxy::ForceSoftwareCapturing` to access its nested enum.

Aditya , do you know if there was a particular reason we chose to forward declare originally? @pxlcoder

Reply:
Not having to pull in `WebPageProxy.h` is a plus for build times.

Reply:
We prefer forward declaring types in headers when we can.

Reply:
I think we should not use a nested enum. It's highly preferable to avoid adding an include to WebPageProxy.

Reply:
OK. Do you think it'll be better to revert this file back to where I had `takeViewSnapshotAlwaysUsingNonHardwareCapturePath` rather than an overload, keeping the nested enum? Otherwise where are you thinking I should put that enum?

Reply:
How about moving to `ViewSnapshotStore.h`?